### DISCUSSION TITLE
permanent_impersonate_prefixes and more

### DISCUSSION THREAD

**Discussion by ecub** [innightwolfsleep/llm_telegram_bot#72] [Q&A]
1) Why does the "Reset" button not reset permanent_impersonate_prefixes?

2) Also, is it possible to somehow limit the number of requests per second to the bot from 1 user?

3) About TBC. Bot works fine in group chats, but I would like to see the ability to turn it off completely in them

Thx

**Comment by innightwolfsleep** [comment]
1. It is because there is no name1 in characters files... This is a bug. I will fix it.
2. This is possible. But is it necesary? I will think about this.
3. Okay, will try to add restriction for group chat. Also, I see request from users to add more complicated user policy. I this will be addded in next big update.

**Reply by ecub** [reply]
The restriction seems necessary because when I send about 5 messages in a row per second, the queue breaks down and the bot starts hallucinating. Fool-proof, or something.

I also thought it would be interesting to see the ability to hide a character's name in the dialog if btn_char_list and load = 0

**Reply by innightwolfsleep** [reply]
added restriction one message per user each 15 sec
